src/useMap.ts

```diff
@@ -0,0 +1,83 @@
+import useForceUpdate from './useForceUpdate'
+import { useRef } from 'react'
+import useWillUnmount from './useWillUnmount'
+
+export class ObservableMap<K, V> extends Map<K, V> {
+  private readonly listeners = new Set<Function>()
```

Comment:
You can probably just take a single, required listener in the ctor. No need to track multiple listeners or deal with empty

Reply:
yeah, tbh i didn't want to type the ctor, then we need to keep it up to date with whatever Set/Map take, which depends on on the `lib` setup in your tsconfig

Reply:
still don't need to support multiple listeners :p